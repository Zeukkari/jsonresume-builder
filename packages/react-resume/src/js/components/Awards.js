import React from 'react'

import { Box } from 'grommet'

import {
  ResumeSection,
  ResumeHeading,
  ResumeSubHeading,
  ResumeParagraph,
  ResumeDateSection,
  SectionTitle,
  formatDate,
} from './common'

const Awards = ({ awards }) => {
  const renderItems = awards.map(item => {
    return (
      <Box key={item.title}>
        <ResumeHeading>{item.title}</ResumeHeading>
        <ResumeSubHeading>{item.awarder}</ResumeSubHeading>
        <ResumeDateSection>{formatDate(item.date)}</ResumeDateSection>
        <ResumeParagraph>{item.summary}</ResumeParagraph>
      </Box>
    )
  })

  return (
    <Box margin="none" pad="none" border="all">
      <ResumeSection>
        <SectionTitle>Awards</SectionTitle>
        <Box>{renderItems}</Box>
      </ResumeSection>
    </Box>
  )
}

Awards.defaultProps = {
  awards: [],
}

export { Awards }
