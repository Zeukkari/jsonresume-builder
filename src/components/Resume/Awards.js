import React from 'react';

import {
  Box,
  DataTable,
  Text,
} from 'grommet';


const columns = [
  {
    property: "title",
    header: <Text>Award</Text>,
    primary: true,
  },
  {
    property: "date",
    header: "Date"
  },
  {
    property: "awarder",
    header: "Company"
  },
  {
    property: "summary",
    header: "Description"
  }
];

const Awards = ({ awards }) => {
  const renderItems = awards;

  return(
    <Box align="center" pad="large">
      <DataTable columns={columns} data={renderItems} />
    </Box>
  )
};

Awards.defaultProps = {
  language: []
}

export default Awards;