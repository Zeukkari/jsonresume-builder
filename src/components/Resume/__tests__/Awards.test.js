import React from 'react';
import renderer from 'react-test-renderer';

import Awards from '../Awards';

import defaultResume from '../../../util/defaultResume';


test('renders awards section', () => {
  const data = defaultResume.awards;
  const component = renderer.create(
    <Awards awards={data} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});