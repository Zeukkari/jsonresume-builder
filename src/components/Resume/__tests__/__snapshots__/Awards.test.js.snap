// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders awards section 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 LVeHq"
>
  <div
    className="StyledBox-sc-13pk1d4-0 jgwjva"
  >
    <h1
      className="StyledHeading-sc-1rdh4aw-0 jVpYfH"
    >
       Awards 
    </h1>
  </div>
  <div
    className="StyledGrommet-sc-19lkkz7-0 lkVXQD"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 jgwjva"
    >
      <table
        className="StyledDataTable-xrlyjm-0 kTrXKK"
      >
        <thead
          className="StyledDataTable__StyledDataTableHeader-xrlyjm-3 bAIczD StyledTable__StyledTableHeader-sc-1m3u5g-4 dHaGyd"
        >
          <tr
            className="StyledDataTable__StyledDataTableRow-xrlyjm-1 ezpFws StyledTable__StyledTableRow-sc-1m3u5g-2 kXgizX"
          >
            <th
              className="StyledTable__StyledTableCell-sc-1m3u5g-0 hNAgVv"
              scope="col"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 folWQp"
              >
                <span
                  className="StyledText-sc-1sadyjn-0 fWSbXS"
                >
                  Award
                </span>
              </div>
            </th>
            <th
              className="StyledTable__StyledTableCell-sc-1m3u5g-0 hNAgVv"
              scope="col"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 folWQp"
              >
                <span
                  className="StyledText-sc-1sadyjn-0 fWSbXS"
                >
                  Date
                </span>
              </div>
            </th>
            <th
              className="StyledTable__StyledTableCell-sc-1m3u5g-0 hNAgVv"
              scope="col"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 folWQp"
              >
                <span
                  className="StyledText-sc-1sadyjn-0 fWSbXS"
                >
                  Company
                </span>
              </div>
            </th>
            <th
              className="StyledTable__StyledTableCell-sc-1m3u5g-0 hNAgVv"
              scope="col"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 folWQp"
              >
                <span
                  className="StyledText-sc-1sadyjn-0 fWSbXS"
                >
                  Description
                </span>
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          className="StyledDataTable__StyledDataTableBody-xrlyjm-2 hbZQKY"
        >
          <tr
            className="StyledDataTable__StyledDataTableRow-xrlyjm-1 ezpFws"
          >
            <th
              className="StyledTable__StyledTableCell-sc-1m3u5g-0 ezepRo"
              scope="row"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 eAAZJM"
              >
                <span
                  className="StyledText-sc-1sadyjn-0 dndifk"
                >
                  Award
                </span>
              </div>
            </th>
            <td
              className="StyledTable__StyledTableCell-sc-1m3u5g-0 ezepRo"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 eAAZJM"
              >
                <span
                  className="StyledText-sc-1sadyjn-0 fWSbXS"
                >
                  2014-11-01
                </span>
              </div>
            </td>
            <td
              className="StyledTable__StyledTableCell-sc-1m3u5g-0 ezepRo"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 eAAZJM"
              >
                <span
                  className="StyledText-sc-1sadyjn-0 fWSbXS"
                >
                  Company
                </span>
              </div>
            </td>
            <td
              className="StyledTable__StyledTableCell-sc-1m3u5g-0 ezepRo"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 eAAZJM"
              >
                <span
                  className="StyledText-sc-1sadyjn-0 fWSbXS"
                >
                  There is no spoon.
                </span>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
